scrapy/http/response/__init__.py

```diff
@@ -75,3 +76,8 @@ def replace(self, *args, **kwargs):
             kwargs.setdefault(x, getattr(self, x))
         cls = kwargs.pop('cls', self.__class__)
         return cls(*args, **kwargs)
+
+    def urljoin(self, uri):
+        """Join this Response's url with a possible relative url to form an
+        absolute interpretation of the latter."""
+        return urlparse.urljoin(self.url, uri)
```

Comment:
No, I mean extracting the base url from the response body using `scrapy.utils.response.get_base_url()` instead of joining directly with `self.url`.